packages/javascript-sdk/src/device-client/services/index.ts

```diff
@@ -44,70 +51,91 @@ export const deviceService = ({ baseUrl, realmPath }: { baseUrl: string; realmPa
+          url: `/json/realms/${realm}/users/${userId}/devices/2fa/push/${device.uuid}`,
           method: 'DELETE',
           body: {},
         }),
       }),
 
       // webauthn devices
-      getWebAuthnDevices: builder.query<WebAuthnDevicesResponse, WebAuthnQuery>({
+      getWebAuthnDevices: builder.query<GeneralResponse<WebAuthnDevice[]>, WebAuthnQuery>({
         query: ({ realm = realmPath, userId }) =>
           `/json/realms/${realm}/users/${userId}/devices/2fa/webauthn?_queryFilter=true`,
       }),
       updateWebAuthnDeviceName: builder.mutation<
         UpdatedWebAuthnDevice,
-        WebAuthnQueryWithUUID & WebAuthnBody
+        WebAuthnQuery & { device: WebAuthnDevice }
       >({
-        query: ({ realm = realmPath, userId, ...device }) => ({
+        query: ({ realm = realmPath, userId, device }) => ({
           url: `/json/realms/${realm}/users/${userId}/devices/2fa/webauthn/${device.uuid}`,
           method: 'PUT',
-          body: device satisfies WebAuthnBody,
+          body: device,
         }),
       }),
       deleteWebAuthnDeviceName: builder.mutation<
-        WebAuthnDevice,
-        WebAuthnQueryWithUUID & WebAuthnBody
+        UpdatedWebAuthnDevice,
+        WebAuthnQuery & { device: UpdatedWebAuthnDevice | WebAuthnDevice }
       >({
-        query: ({ realm = realmPath, userId, ...device }) => ({
+        query: ({ realm = realmPath, userId, device }) => ({
           url: `/json/realms/${realm}/users/${userId}/devices/2fa/webauthn/${device.uuid}`,
           method: 'DELETE',
-          body: device satisfies WebAuthnBody,
+          body: device,
         }),
       }),
-      getBoundDevices: builder.mutation<DeviceResponse, BindingDeviceQuery>({
+      getBoundDevices: builder.mutation<GeneralResponse<Device[]>, GetBoundDevicesQuery>({
         query: ({ realm = realmPath, userId }) =>
           `/json/realms/${realm}/users/${userId}/devices/2fa/binding?_queryFilter=true`,
       }),
       updateBindingDeviceName: builder.mutation<Device, BindingDeviceQuery>({
-        query: ({ realm = realmPath, userId, ...device }) => ({
+        query: ({ realm = realmPath, userId, device }) => ({
           url: `/json/realms/root/realms/${realm}/users/${userId}/devices/2fa/binding/${device.uuid}`,
           method: 'PUT',
-          body: device satisfies Device,
+          body: device,
         }),
       }),
-      deleteBindingDevice: builder.mutation<Device, BindingDeviceQuery>({
-        query: ({ realm = realmPath, userId, ...device }) => ({
+      deleteBindingDevice: builder.mutation<GeneralResponse<Device>, BindingDeviceQuery>({
+        query: ({ realm = realmPath, userId, device }) => ({
           url: `/json/realms/root/realms/${realm}/users/${userId}/devices/2fa/binding/${device.uuid}`,
           method: 'DELETE',
           body: device satisfies Device,
         }),
       }),
+      getDeviceProfile: builder.query<GeneralResponse<ProfileDevice[]>, GetProfileDevices>({
```

Comment:
Are we querying for a single profile here, or are we querying for all of the user's profiles. I would think all, so I'd expect the name to be plural: `getDeviceProfiles`.